### PR TITLE
refresh cluster from api on update

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Refresh cluster from api on update

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/controller/cluster/cluster_controller.go [hunk 1]
```diff
@@ -264,12 +264,11 @@ func (cc *Controller) enqueue(cluster *kubermaticv1.Cluster) {
 
 func (cc *Controller) updateCluster(name string, modify func(*kubermaticv1.Cluster)) (updatedCluster *kubermaticv1.Cluster, err error) {
 	err = retry.RetryOnConflict(retry.DefaultBackoff, func() (err error) {
-		//Get latest version from cache
-		cacheCluster, err := cc.clusterLister.Get(name)
+		//Get latest version from api
+		currentCluster, err := cc.kubermaticClient.KubermaticV1().Clusters().Get(name, metav1.GetOptions{})
```
Comment (alvaroaleman) [inline api/pkg/controller/cluster/cluster_controller.go]: Uhm what, why does the lister not contain the most up-to-date version?

**Review by alvaroaleman — CHANGES_REQUESTED** [review]

File: api/pkg/controller/cluster/cluster_controller.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/cluster/cluster_controller.go]: It seems in some kubernetes setups (like AKS) the apiserver takes more than 30sec for a watch-event. As we have a exponential backoff here, an update statement leads to a very long waiting time

Comment (alvaroaleman) [reply api/pkg/controller/cluster/cluster_controller.go]: Not sure I understand. So creating the watch takes 30s? Why does that matter? The lister is supposed to always contain the most up-to-date version

Comment (mrIncompetent) [reply api/pkg/controller/cluster/cluster_controller.go]: The events delivered from the watch endpoint take very long to be populated by the apiserver

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**